front/temporal/labs/activities.ts

```diff
@@ -189,64 +129,53 @@ export async function processGoogleDriveTranscriptActivity(
   if (!user) {
     localLogger.error(
       {},
-      "[processGoogleDriveTranscriptActivity] User not found. Stopping."
+      "[processTranscriptActivity] User not found. Stopping."
     );
     return;
   }
 
   localLogger.info(
     {},
-    "[processGoogleDriveTranscriptActivity] Starting processing of file "
+    "[processTranscriptActivity] Starting processing of file "
   );
 
   const hasExistingHistory =
     await transcriptsConfiguration.fetchHistoryForFileId(fileId);
   if (hasExistingHistory) {
     localLogger.info(
-      "[processGoogleDriveTranscriptActivity] History record already exists. Stopping."
+      "[processTranscriptActivity] History record already exists. Stopping."
     );
     return;
   }
 
-  const googleAuth = await getGoogleAuthFromUserTranscriptsConfiguration(auth);
-  const drive = googleapis.google.drive({ version: "v3", auth: googleAuth });
-
-  const metadataRes = await drive.files.get({
-    fileId: fileId,
-    fields: "name",
-  });
-
-  const contentRes = await drive.files.export({
-    fileId: fileId,
-    mimeType: "text/plain",
-  });
-
-  if (contentRes.status !== 200) {
-    localLogger.error(
-      { error: contentRes.statusText },
-      "Error exporting Google document."
-    );
+  let transcriptTitle = "";
+  let transcriptContent = "";
 
-    throw new Error(
-      `Error exporting Google document. status_code: ${contentRes.status}. status_text: ${contentRes.statusText}`
+  if (transcriptsConfiguration.provider == "google_drive") {
```

Comment:
switch case with a default: assertNever() to make sure we update this code when we add a provider.